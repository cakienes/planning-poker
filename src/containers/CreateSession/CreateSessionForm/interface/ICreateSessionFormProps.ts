export default interface ICreateSessionFormProps {
    handleSubmit?: (values: any) => void;
}
