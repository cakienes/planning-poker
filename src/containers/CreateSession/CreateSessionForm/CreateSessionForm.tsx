import React from 'react';
import { Field, reduxForm } from 'redux-form';
import Button from '../../../components/Button/Button';
import FormFields from '../../../components/FormFields/FormFields';
import { maxLength } from '../../../helper/Normalize';
import { onlyNumber, required } from '../../../helper/Validation';
import './CreateSessionForm.scss';
import ICreateSessionFormProps from './interface/ICreateSessionFormProps';

export const CreateSessionForm = (props: ICreateSessionFormProps) => {
    return (
        <form onSubmit={props.handleSubmit}>
            <div className="row">
                <div className="col-6">
                    <Field
                        name="sessionName"
                        label="Session Name: "
                        component={FormFields}
                        type="text"
                        validate={required}
                        normalize={maxLength(200)}
                    />
                </div>
                <div className="col-6">
                    <Field
                        name="numberOfVoters"
                        label="Number of Voters: "
                        component={FormFields}
                        type="number"
                        validate={[required, onlyNumber]}
                    />
                </div>
            </div>
            <div className="row">
                <div className="col">
                    <Field name="optionalUrl" label="URL (Optional): " component={FormFields} type="text" />
                </div>
            </div>
            <div className="row">
                <div className="col">
                    <Field
                        name="userStories"
                        label="Paste your story list (Each line will be converted as a story): "
                        component={FormFields}
                        type="textarea"
                        validate={required}
                    />
                </div>
            </div>
            <div className="row">
                <div className="col">
                    <div className="buttonHolder">
                        <Button label="Start Session" type="submit" />
                    </div>
                </div>
            </div>
        </form>
    );
};

export default reduxForm({
    form: 'createSessionForm',
})(CreateSessionForm);
