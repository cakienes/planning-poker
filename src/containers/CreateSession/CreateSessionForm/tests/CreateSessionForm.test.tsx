import React from 'react';
import { createRenderer, ShallowRenderer } from 'react-test-renderer/shallow';
import { CreateSessionForm } from '../CreateSessionForm';
import ICreateSessionFormProps from '../interface/ICreateSessionFormProps';

describe('CreateSessionForm.tsx', () => {
    let renderer: ShallowRenderer;
    const handleSubmitMock = jest.fn();

    let createSessionFormProps: ICreateSessionFormProps = {
        handleSubmit: handleSubmitMock,
    };

    beforeEach(() => {
        renderer = createRenderer();
        createSessionFormProps = {
            handleSubmit: handleSubmitMock,
        };
    });

    it('renders correctly CreateSessionForm', async () => {
        expect(renderer.render(<CreateSessionForm {...createSessionFormProps} />)).toMatchSnapshot();
    });
});
