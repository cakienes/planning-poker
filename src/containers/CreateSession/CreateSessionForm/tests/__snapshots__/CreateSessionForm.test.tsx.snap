// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateSessionForm.tsx renders correctly CreateSessionForm 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <ForwardRef
        component={[Function]}
        label="Session Name: "
        name="sessionName"
        normalize={[Function]}
        type="text"
        validate={[Function]}
      />
    </div>
    <div
      className="col-6"
    >
      <ForwardRef
        component={[Function]}
        label="Number of Voters: "
        name="numberOfVoters"
        type="number"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <ForwardRef
        component={[Function]}
        label="URL (Optional): "
        name="optionalUrl"
        type="text"
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <ForwardRef
        component={[Function]}
        label="Paste your story list (Each line will be converted as a story): "
        name="userStories"
        type="textarea"
        validate={[Function]}
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="buttonHolder"
      >
        <Button
          label="Start Session"
          type="submit"
        />
      </div>
    </div>
  </div>
</form>
`;
